core/src/main/scala/cats/Foldable.scala

```diff
@@ -152,6 +152,26 @@ import simulacrum.typeclass
   def size[A](fa: F[A]): Long = foldMap(fa)(_ => 1)
 
   /**
+    * Get the element at the index of the `Foldable`.
+    *
+    * The default implementation of this is based on `foldLeft`, and thus will
+    * always fold across the entire structure.
+    */
+  def get[A](fa: F[A])(idx: Long): Option[A] =
+    foldLeft(fa, (0L, Option.empty[A])) { (b, a) =>
+      b._1 match {
+        case i if i == idx  => (i + 1, Some(a))
+        case i if i < idx => (i + 1, None)
+        case _ => b
+      }
+    }._2
+
+  /**
+    * Get the element at the index of the `Foldable`
+    */
+  def apply[A](fa: F[A])(idx: Long): A = get(fa)(idx).get
```

Comment:
I would agree for functions which aren't obviously unsafe - the `unsafeRun` in the `Stream` in fs2 comes to mind. However, for getting any index or key in a `Map` or `List` using `apply` in scala is already implied shorthand for `get.get`? 

Since I would argue that users should already be familiar with this behaviour, I can't see much benefit in calling it `getUnsafe`.

I don't feel particularly strongly enough to not change it, but I don't think to signal it's an unsafe method to users is a good enough reason.